velox/serializers/UnsafeRowSerializer.cpp

```diff
@@ -0,0 +1,158 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#include "velox/serializers/UnsafeRowSerializer.h"
+#include "velox/row/UnsafeRowDeserializer.h"
+#include "velox/row/UnsafeRowDynamicSerializer.h"
+
+namespace facebook::velox::serializer::spark {
+
+void UnsafeRowVectorSerde::estimateSerializedSize(
+    VectorPtr vector,
+    const folly::Range<const IndexRange*>& ranges,
+    vector_size_t** sizes) {
+  for (auto& range : ranges) {
+    for (auto i = range.begin; i < range.begin + range.size; ++i) {
+      size_t rowSize = velox::row::UnsafeRowDynamicSerializer::getSize(
+          vector->type(), vector, i);
+      (*sizes)[i] += rowSize;
```

Comment:
Yes. I'll make a comment in the API.

```
void PartitionedOutput::estimateRowSizes() {
  auto numInput = input_->size();
  std::fill(rowSize_.begin(), rowSize_.end(), 0);
  for (int i = 0; i < output_->childrenSize(); ++i) {
    VectorStreamGroup::estimateSerializedSize(
        output_->childAt(i),
        folly::Range(topLevelRanges_.data(), numInput),
        sizePointers_.data());
  }
}
```

Reply:
Since we don't need this method for the test and we don't plan to use this in prod, I'm going to remove the implementation and use VELOX_UNSUPPORTED() instead.